docs/HyperIndex/Guides/price-data.md

```diff
@@ -0,0 +1,273 @@
+:::
+
+```typescript
+const SUBGRAPH_API_KEY = process.env.SUBGRAPH_API_KEY;
+
+async function fetchEthPrice(block: number): Promise<any> {
+    const response = await fetch(`https://gateway.thegraph.com/api/${SUBGRAPH_API_KEY}/subgraphs/id/2LHovKznvo8YmKC9ZprPjsYAZDCc4K5q4AYz8s3cnQn1`, {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+            query: `{ pool(id: "0xf52b4b69123cbcf07798ae8265642793b2e8990c", block: {number: ${block}}) { token1Price token0Price } }`,
+            operationName: "Subgraphs",
+            variables: {}
+        })
+    });
+    const data = await response.json() as any;
+    return Number(data.data.pool.token0Price);
+}
+
+export default fetchEthPrice;
+```
+
+### Offchain API considerations
+
+- **Accuracy**: This is the most accurate and reliable to get price data in your indexer as you can make an rpc call to get the very latest USD price. Some APIs, such as CoinGecko, aggregate price data from many different sources so price feed manipulation is highly unlikely. The only real risk you run is if the API goes down or sufferes manipulation, which is very rare for reputable APIs.
+- **Centralization**: Offchain APIs are typically controlled by a single entity meaning they are very centralized. This means that you have to trust the API to provide accurate and reliable price data. However, as mentioned, reputable APIs are very unlikely to manipulate price data. This is not a concern for most users.
+- **Data availability**: Offchain APIs such as CoinGecko provide price feeds for the majority of tokens on the blockchain with real volume. However, they may not have price data for newer, low marketcap tokens. In that case, you may have to rely on dex pools or oracles to get price data. Note that offchain APIs require an rpc call every time you need price data which will **significantly** slow down your indexer. RPC calls will change your indexer from a fast, event-based indexer to a slow, rpc-based indexer that may take hours to run.
+
+Due to the slow nature of RPC calls and the fact that you may need a paid subscription, only use this method if accuracy is of the utmost importance. This will help to speed up your indexer and reduce the cost of rpc calls.
+
+## Event handler
+
+To compare these methods, we will compare the values of the USDB/WETH pool on Uniswap v3 after every `Mint` event.
+
+`npx envio init`
+
+- For our oracle information, `0x709944a48cAf83535e43471680fDA4905FB3920a` contract and the `UpdatedBeaconSetWithBeacons` event.
+- For our dex pool and tvl information, `0xf52B4b69123CbcF07798AE8265642793b2E8990C` contract and the `Swap` and `Mint` event.
+
+> config.yaml
+
+```yaml
+# yaml-language-server: $schema=./node_modules/envio/evm.schema.json
+name: envio-indexer
+networks:
+- id: 81457
+  start_block: 0
```

Comment:
Can set the start block here if you don't want to calculate before block ~1100000